crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -311,10 +312,17 @@ async fn payment_response_update_tracker<F: Clone, T: types::Capturable>(
                     (Some((multiple_capture_data, capture_update_list)), None)
                 }
                 None => {
-                    let status = match err.status_code {
-                        500..=511 => storage::enums::AttemptStatus::Pending,
-                        _ => storage::enums::AttemptStatus::Failure,
-                    };
+                    let flow_name = core_utils::get_flow_name::<F>()?;
+                    let status =
+                        // mark previous attempt status for technical failures in PSync flow
+                        if flow_name == "PSync" {
+                            payment_data.payment_attempt.status
```

Comment:
here we are not allowing the status to be changed in case of genuine error too, this can be handled by checking the status code and then updating status